server_tests/lib/schema.js

```diff
@@ -0,0 +1,1169 @@
+    }
+
+    this.state = {
+        type: "object",
+        additionalProperties: false,
+        patternProperties: {
+            "^_": {},
+        },
+        required: ["fingerprint", "state"],
+        properties: {
+            fingerprint: {
+                type: "string",
+            },
+            state: {
+                type: "object",
+                additionalProperties: false,
+                patternProperties: {
+                    "^_": {},
+                },
+                required: ["sessionId", "batteryLevel"],
+                properties: {
+                    sessionId: {
+                        type: "string",
+                    },
+                    batteryLevel: {
+                        type: "number",
+                        minimum: 0,
+                        maximum: this.bubl ? 100 : 1,
+                    },
+                    storageChanged: {
+                        type: "boolean",
+                    },
+                    storageUri: {
+                        type: "string",
+                    },
+                    _bublCaptureMode: {
+                        type: "string",
+                    },
+                    _bublCaptureMode2: {
+                        type: "string",
+                    },
+                    _bublCommands: {
+                        type: "array",
+                        items: this.status,
+                    },
+                    _bublCharging: {
+                        type: "boolean",
+                    },
+                    _bublChargingSufficientPower: {
+                        type: "boolean",
+                    },
+                    _bublSdInserted: {
+                        type: "boolean",
+                    },
+                    _bublLatestCapture: this._commandResults1[this.names.commandListImages].properties.entries.items,
+                    _bublLatestCapture2: this._commandResults2[this.names.commandListFiles].properties.entries.items,
+                },
+            },
+        },
+    };
+
+    this.checkForUpdates = {
+        type: "object",
+        additionalProperties: false,
+        patternProperties: {
+            "^_": {},
+        },
+        required: ["stateFingerprint", "throttleTimeout"],
+        properties: {
+            stateFingerprint: {
+                type: "string",
+            },
+            throttleTimeout: {
+                type: "number",
+                minimum: 0,
+            },
+        },
+    };
+
+    this.bublPoll = {
+        type: "object",
+        additionalProperties: false,
+        patternProperties: {
+            "^_": {},
+        },
+        required: ["fingerprint", "throttleTimeout", "command"],
+        properties: {
+            fingerprint: {
+                type: "string",
+            },
+            throttleTimeout: {
+                type: "number",
+                minimum: 0,
+            },
+            command: {
```

Comment:
There's already a schema for this, we can validate more than just "is object":

    command: this.status,